package.json

```diff
@@ -21,24 +21,77 @@
     "video.js": "5.x"
   },
   "scripts": {
-    "build": "mkdirp dist && cp src/Youtube.js dist/Youtube.js && uglifyjs src/Youtube.js -o dist/Youtube.min.js",
+    "prebuild": "npm run clean",
+    "build": "npm-run-all -p build:*",
+    "build:js": "npm-run-all mkdirs build:js:babel build:js:browserify build:js:bannerize build:js:uglify",
+    "build:js:babel": "babel src -d es5",
+    "build:js:bannerize": "bannerize dist/Youtube.js --banner=scripts/banner.ejs",
+    "build:js:browserify": "browserify . -s Youtube -o dist/Youtube.js",
+    "build:js:uglify": "uglifyjs dist/Youtube.js --comments --mangle --compress -o dist/Youtube.min.js",
+    "build:test": "npm-run-all mkdirs build:test:browserify",
+    "build:test:browserify": "browserify `find test -name '*.specs.js'` -t babelify -o dist-test/Youtube.js",
     "lint": "jshint .",
-    "test": "karma start --single-run",
+    "clean": "rm -rf dist dist-test es5",
+    "mkdirs": "mkdir -p dist dist-test es5",
+    "prepublish": "npm run build",
+    "prestart": "npm-run-all -p docs build",
+    "dev": "npm-run-all -p start watch",
+    "start": "npm-run-all -p start:serve",
+    "start:serve": "babel-node --optional es7.functionBind,es7.classProperties,es7.decorators  scripts/server.js",
+    "pretest": "npm-run-all build:test",
+    "test": "karma start --single-run test/karma/detected.js",
+    "test:chrome": "npm run pretest && karma start test/karma/chrome.js",
+    "test:firefox": "npm run pretest && karma start test/karma/firefox.js",
+    "test:ie": "npm run pretest && karma start test/karma/ie.js",
+    "test:safari": "npm run pretest && karma start test/karma/safari.js",
+    "watch": "npm run mkdirs && npm-run-all -p watch:*",
+    "watch:js": "watchify src/Youtube.js -t babelify -v -o dist/Youtube.js",
     "validate": "npm ls"
   },
   "pre-commit": [
     "lint"
   ],
+  "browserify": {
+    "transform": [
+      "browserify-shim"
+    ]
+  },
+  "browserify-shim": {
+    "qunit": "global:QUnit",
+    "sinon": "global:sinon",
+    "video.js": "global:videojs"
+  },
+  "babel": {
+    "optional": [
+      "es7.functionBind",
```

Comment:
I don't think we should be using experimental features of the language and definitely not stuff that aren't on track to be added to the language.